src/devices/Mpu9250/Mpu9250.cs

```diff
@@ -0,0 +1,983 @@
+        /// </summary>
+        public void Dispose()
+        {
+            if (_autoDispose)
+            {
+                _i2cDevice?.Dispose();
+                _i2cDevice = null;
+            }
+        }
+
+        #endregion
+
+        #region FIFO
+
+        /// <summary>
+        /// Get the number of elements to read from the FIFO
+        /// </summary>
+        public uint FifoCount
+        {
+            get
+            {
+                Span<byte> rawData = stackalloc byte[2] { 0, 0 };
+                ReadByteArray(Register.FIFO_COUNTH, rawData);
+                return BinaryPrimitives.ReadUInt16BigEndian(rawData);
+            }
+        }
+
+        /// <summary>
+        /// Get or set the FIFO modes
+        /// </summary>
+        public FifoModes FifoModes
+        {
+            get { return (FifoModes)(ReadByte(Register.FIFO_EN)); }
+            set
+            {
+                if (value != FifoModes.None)
+                {
+                    // Make sure the FIFO is enbalbe
+                    var usrCtl = (UserControls)ReadByte(Register.USER_CTRL);
+                    usrCtl = usrCtl | UserControls.FIFO_RST;
+                    WriteRegister(Register.USER_CTRL, (byte)usrCtl);
+                }
+                else
+                {
+                    // Deactivate FIFO
+                    var usrCtl = (UserControls)ReadByte(Register.USER_CTRL);
+                    usrCtl = usrCtl & ~UserControls.FIFO_RST;
+                    WriteRegister(Register.USER_CTRL, (byte)usrCtl);
+                }
+                WriteRegister(Register.FIFO_EN, (byte)value);
+            }
+        }
+
+        /// <summary>
+        /// Read data in the FIFO, read as many data as the size of readData byte span
+        /// You should read the number of data availavle in the FifoCount property then
+        /// read them here.
+        /// You will read only data you have selected in FifoModes.
+        /// Data are in the order of the Register from 0x3B to 0x60.
+        /// ACCEL_XOUT_H and ACCEL_XOUT_L
+        /// ACCEL_YOUT_H and ACCEL_YOUT_L 
+        /// ACCEL_ZOUT_H and ACCEL_ZOUT_L
+        /// TEMP_OUT_H and TEMP_OUT_L
+        /// GYRO_XOUT_H and GYRO_XOUT_L
+        /// GYRO_YOUT_H and GYRO_YOUT_L
+        /// GYRO_ZOUT_H and GYRO_ZOUT_L
+        /// EXT_SENS_DATA_00 to EXT_SENS_DATA_24
+        /// </summary>
+        /// <param name="readData">Data which will be read</param>
+        public void ReadFifo(Span<byte> readData)
+        {
+            ReadByteArray(Register.FIFO_R_W, readData);
+        }
+
+        #endregion
+
+        #region Calibration and tests
+
+        /// <summary>
+        /// Perform full calibration the gyroscope and the accelerometer
+        /// It will automatically adjust as well the offset stored in the device
+        /// The result bias will be stored in the AcceloremeterBias and GyroiscopeBias
+        /// </summary>
+        public void CalibrateGyroscopeAccelerometer()
```

Comment:
This method seems to be an extensive process. Can we add a comment pointing to the page on the manual describing what needs to be done in case this needs to be changed in the future?

Reply:
There is nothing specific in the documentation. The way calibration is done is using the same method as for all gyroscope and accelerometers. 
Basic idea is to find the bias by measuring couple of times and storing back the bias.
The code itself is documented almost line by line to explain every steps and what is done.